### PR TITLE
Circleci tests

### LINKED ISSUE #844: Investigate failing `should generate all apis`test
This test is failing. Usually happens when there's some kind of invalid discovery doc pushed out to apiary.  Unit tests with external dependencies that change outside of our code :(

### PR DESCRIPTION
**Pull request by alexander-fenster**
(a) new CircleCI config
(b) updated some packages

Fixes #844 and possibly #845 as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,24 +93,25 @@
     "@types/minimist": "1.2.0",
     "@types/mkdirp": "0.5.2",
     "@types/mocha": "2.2.45",
+    "@types/nock": "^9.1.2",
```
Comment (JustinBeckwith) [inline package.json]: lets update nock in another PR

File: package.json [hunk 2]
```diff
@@ -93,24 +93,25 @@
...
     "@types/node": "8.5.2",
-    "@types/nock": "^8.2.1",
     "@types/nunjucks": "^3.0.0",
     "@types/pify": "^3.0.0",
     "@types/qs": "^6.5.1",
     "@types/uuid": "^3.4.3",
     "axios": "^0.17.1",
     "clang-format": "^1.1.1",
+    "codecov": "^3.0.0",
```
Comment (JustinBeckwith) [inline package.json]: It looks like you added codecov, but I don't see where you use it

**Comment by JustinBeckwith** [comment]
Oh yeah, almost forgot - can you include the README.md badge change here as well?

**Comment by alexander-fenster** [comment]
nock update is the reason why tests fail on node4 and node6. They hang in https request that nock was supposed to handle. I did not dig deeper but updating seems to help (CI proves that).

As for codecov, it’s being run in Circle and results will be reported to codecov.io when enabled there.

Should I still move nock upgrade to a separate PR?

**Comment by JustinBeckwith** [comment]
Nah, it's fine.  Happy the test fix was this easy :)

**Comment by JustinBeckwith** [comment]
I'd love to merge this, but I have no idea how to get the travis-ci check to go away  😂  I deleted the TravisCI integration from the project, and I tried poking it with another commit on your branch, but it's still there.

**Comment by alexander-fenster** [comment]
I reconfigured master branch protection to require circleci checks to pass and removing Travis.

**Pull request merged**